coalib/core/Core.py

```diff
@@ -212,8 +212,8 @@ def _schedule_bears(self, bears):
         """
         for bear in bears:
             if self.dependency_tracker.get_dependencies(
-                    bear):  # pragma: no cover
-                logging.warning(
+                    bear):
+                raise RuntimeError(  # Unreachable code
```

Comment:
The problem is currently that the core doesn't exit properly with this exception, as this function gets also run on the processes and following cleanups aren't executed.